Source/3rdParty/mupen64plus-input-raphnetraw/projects/unix/Makefile

```diff
@@ -141,6 +141,14 @@ ifeq ($(OS), OSX)
   endif
 endif
 
+# test for essential build dependencies
+ifeq ($(origin PKG_CONFIG), undefined)
+  PKG_CONFIG = $(CROSS_COMPILE)pkg-config
+  ifeq ($(shell which $(PKG_CONFIG) 2>/dev/null),)
```

Comment:
upstream mupen64plus-core also uses `which` there though.

Reply:
Then mupen64plus-core, too, is wrong. :)

My attention was attracted here because @orbea implemented this change for Gentoo and I reviewed that change. I'm not directly familiar with this ecosystem though I am always happy to provide tips such as this one when I'm made aware of it.

Reply:
Informative reading :) https://unix.stackexchange.com/questions/85249/why-not-use-which-what-to-use-then/85250#85250

Reply:
> Then mupen64plus-core, too, is wrong. :)
I'm not sure if I'd consider using `which` to be wrong per se, especially considering how widespread the usage of that command is in general.

It's just a dependency which exists, it's neither right or wrong in my opinion, and changing it for the sake of changing it is possible but I don't feel like it matters much because almost every linux system has `which` available or installed.

Reply:
I have used multiple Linux distros where it wasn't available by default.

It is most commonly seen on Debian specifically because Debian has a policy guide saying that the "which" program which they privately wrote from scratch must be installed in the base system in order to be available in dpkg postinst scripts. And sure enough... Debian is trying to purge the use of "which" so they can finally drop that. :)

People often tend to code for Debian specifically. It's unfortunate. The assumption is that since Debian is the numerically largest number of Linux systems, that writing software for "Debian" is the same as writing software for "Linux".

> It's just a dependency which exists, it's neither right or wrong in my opinion

That's an opinion.

I informed you of a *fact*, however. It is "factual* that it's wrong, and that you cannot even assume that "which" will return success when the command exists and an error when it doesn't. You can't assume it only prints stdout on success. You can't assume its results make sense for your current environment (rather than reading the rc scripts from your interactive shell). You can't assume a lot of things.

It's also quite weird to force people to install a specific dependency in order to query if an entirely different command exists on PATH, since if a user has to manually install something anyway they might as well manually install pkg-config itself. Given the shell can do this built-in... why not use the builtin?

What's the appeal of adding an additional third party dependency that is commonly not installed when the base operating system has a more reliable alternative already?

It's an incredibly simple change, anyway. :shrug:

Reply:
> It's an incredibly simple change, anyway. 🤷

This is only true if you don't consider having to upstream the patch for each project which RMG uses which uses that command, it's time consuming for no real gain.

But if you consider this to be a simple change, then feel free to do so for each upstream project yourself.